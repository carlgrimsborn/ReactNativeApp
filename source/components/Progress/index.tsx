import * as React from 'react';
import { View, Text, ActivityIndicator } from 'react-native';

const Progress = ({amount}: IProps) => {
  return (
    <View>
      <ActivityIndicator size='small' color='#0000ff' />
      <Text>Progress: {amount}</Text>
    </View>
  );
};

interface IProps {
  amount: number;
}

export default Progress;
