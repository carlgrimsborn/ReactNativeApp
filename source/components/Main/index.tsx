import { SocialIcon } from 'react-native-elements';
import React, { Component } from 'react';
import {
  AppRegistry,
  View,
  Image,
  StyleSheet,
  TextInput,
  Text
} from 'react-native';
import About from '../About';
import Progress from '../Progress';
import { Actions } from 'react-native-router-flux';

interface IState {
    userName: string;
    amountClicks: number;
}

export default class Main extends Component<{}, IState> {
  constructor(props: any) {
    super(props);
    this.state = {
      userName: 'Who?',
      amountClicks: 0,
    };
  }
  componentDidMount() {
    Actions.push('login');
  }
  onTextChange(newText: string) {
    this.setState({
      userName: newText,
      amountClicks: this.state.amountClicks + 1,
    });
  }
  render() {
    return (
      <View style={styles.container}>
        <TextInput
          placeholder='who are thee'
          onChangeText={text => this.onTextChange(text)}
        />
        <Text>{this.state.userName}</Text>

        <SocialIcon
          style={{ width: 225 }}
          title='Sign In With Facebook'
          button
          type='facebook'
        />
        <About aboutText='We are a happy people' />
        <Image
          style={{ width: 200, height: 200 }}
          source={require('../../../2-Hot-Home-icon.png')}
        />

        <Progress amount={this.state.amountClicks} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
