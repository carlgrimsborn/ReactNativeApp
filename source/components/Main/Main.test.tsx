/// <reference types="jest"/>
import React from 'react';
import { shallow } from 'enzyme';
import Progress from '../Progress';

import Main from './';

function setup() {
  const props = {
// Här kan ni mocka komponentens props om sådana finns
// typ userName: 'test'
  };

  const enzymeWrapper = shallow(<Main {...props} /> );

  return {
    props,
    enzymeWrapper,
  };
}

describe('components', () => {
  /**
   * Test component rendering. Properties of children might be tested by importing their type:
   *   import { Text } from 'react-native';
   *    const busyProps = enzymeWrapper.find(Text).props();
   *    expect(busyProps.text).toBe('test);
   */
  describe('Main', () => {
    it('should render self and subcomponents', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();

      // Test Progress
      const progress = enzymeWrapper.find(Progress).props();
      expect(progress.amount).toBe(0);

    });
  });
});