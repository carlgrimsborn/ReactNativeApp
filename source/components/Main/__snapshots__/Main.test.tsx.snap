// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Main should render self and subcomponents 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED
          onChangeText={[Function]}
          placeholder="who are thee"
        />,
        <UNDEFINED>
          Who?
        </UNDEFINED>,
        <UNDEFINED
          button={true}
          style={
            Object {
              "width": 225,
            }
          }
          title="Sign In With Facebook"
          type="facebook"
        />,
        <about
          aboutText="We are a happy people"
        />,
        <UNDEFINED
          source={
            Object {
              "testUri": "../../../2-Hot-Home-icon.png",
            }
          }
          style={
            Object {
              "height": 200,
              "width": 200,
            }
          }
        />,
        <Progress
          amount={0}
        />,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeText": [Function],
          "placeholder": "who are thee",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Who?",
        },
        "ref": null,
        "rendered": "Who?",
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "button": true,
          "style": Object {
            "width": 225,
          },
          "title": "Sign In With Facebook",
          "type": "facebook",
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "aboutText": "We are a happy people",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "source": Object {
            "testUri": "../../../2-Hot-Home-icon.png",
          },
          "style": Object {
            "height": 200,
            "width": 200,
          },
        },
        "ref": null,
        "rendered": null,
        "type": undefined,
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "amount": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": undefined,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED
            onChangeText={[Function]}
            placeholder="who are thee"
          />,
          <UNDEFINED>
            Who?
          </UNDEFINED>,
          <UNDEFINED
            button={true}
            style={
              Object {
                "width": 225,
              }
            }
            title="Sign In With Facebook"
            type="facebook"
          />,
          <about
            aboutText="We are a happy people"
          />,
          <UNDEFINED
            source={
              Object {
                "testUri": "../../../2-Hot-Home-icon.png",
              }
            }
            style={
              Object {
                "height": 200,
                "width": 200,
              }
            }
          />,
          <Progress
            amount={0}
          />,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChangeText": [Function],
            "placeholder": "who are thee",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Who?",
          },
          "ref": null,
          "rendered": "Who?",
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "button": true,
            "style": Object {
              "width": 225,
            },
            "title": "Sign In With Facebook",
            "type": "facebook",
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aboutText": "We are a happy people",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "source": Object {
              "testUri": "../../../2-Hot-Home-icon.png",
            },
            "style": Object {
              "height": 200,
              "width": 200,
            },
          },
          "ref": null,
          "rendered": null,
          "type": undefined,
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "amount": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": undefined,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
